### PR TITLE
Add envsubst steps to lab 3

### LINKED ISSUE #140: Add lab / steps for envvarsubst
## Task

Add lab / steps for envvarsubst

## Context

https://github.com/openfaas/faas-cli/pull/626

This has been tracked on Trello since April, but not worked on yet.

I would like a volunteer to put some content together showing how to use the feature. See the docs or README from the CLI for some inspiration.

https://docs.openfaas.com

https://github.com/openfaas/faas-cli

### PR DESCRIPTION
**Pull request by burtonr**
Added an additional optional step to demonstrate the environment
variable substitution feature in the yml files with a real-world
use case as an example scenario

The resolves #140 

Signed-off-by: Burton Rheutan <rheutan7@gmail.com>

### CODE REVIEW

File: lab3.md [hunk 1]
```diff
@@ -438,6 +438,45 @@ You can now type in `faas-cli new --lang node10-express`.
 
 See also: [Function & Template Store](https://github.com/openfaas/store/)
 
+### Variable Substitution in YAML File (optional exercise)
+
+The `.yml` file used to configure the CLI is capable of variable substitution so that you are able to use the same `.yml` file for multiple configurations.
+
+One example of where this can be used is when there are different registries for development and production images. You can use the variable substitution so that local and test environments use the default account, and the CI server can be configured to use the production account.
```
Comment (rgee0) [inline lab3.md]: `One example of where this can be used`

consider used => useful

Comment (burtonr) [reply lab3.md]: Good catch, thank you. I've updated this.

I've also added  a note mentioning that all references to the variable will be replaced throughout the file.

File: lab3.md [hunk 2]
```diff
@@ -438,6 +438,47 @@ You can now type in `faas-cli new --lang node10-express`.
 
 See also: [Function & Template Store](https://github.com/openfaas/store/)
 
+### Variable Substitution in YAML File (optional exercise)
```
Comment (alexellis) [inline lab3.md]: I like this a lot.

I wonder whether we can use it to show how to deploy a function ending in `-dev/-prod` in the function's name field?

The Docker image is a useful part to change, anything else for first-time users?

**Comment by alexellis** [comment]
Thanks for the review + input Richard.

**Pull request merged**